soc/wch/ch32v00x/soc.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024 Michael Hope
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <ch32v003fun.h>
+#include <soc.h>
+
+#define PLL_CSS_HSE_MASK ((uint32_t)0xFEF6FFFF)
+#define HSEBYP_MASK      ((uint32_t)0xFFFBFFFF)
+#define PLLSRC_MASK      ((uint32_t)0xFFFEFFFF)
+
+static void soc_ch32v003_delay_start(void)
+{
+	for (volatile int i = 0; i < 500000; i++) {
+		__asm("nop");
+		__asm("nop");
+		__asm("nop");
+		__asm("nop");
+		__asm("nop");
+	}
+}
```

Comment:
Don't know the internals of this but 250,000 NOPs at start up seems like a long time of doing nothing without a comment as to why?

Reply:
Perhaps they are doing this to allow ample time to connect a debugger? Or do switch to program mode on PD1? It's definitely not required though

Reply:
For context, I wrote the original version of this, and @kholia has taken it over and is upstreaming it (thanks kholia!).

This and the corresponding check in the GPIO driver can be removed. The SWD is on a general purpose I/O pin so, if you set the pinctrl incorrectly, then you can prevent the debugger from working and effectively brick the board. This spin is to give the debugger a chance to connect before the pinctrl code runs.

A better way is to use minichlink's `-u` command (hi @cnlohr !) which also unbricks the chip.

Reply:
This should be fixed now - thanks!